src/processTargets/modifiers/scopeHandlers/OneOfScopeHandler.ts

```diff
@@ -0,0 +1,113 @@
+import { minBy } from "lodash";
+import { Position, Range, TextEditor } from "vscode";
+import { getScopeHandler } from ".";
+import {
+  Direction,
+  OneOfScopeType,
+} from "../../../typings/targetDescriptor.types";
+import NotHierarchicalScopeError from "./NotHierarchicalScopeError";
+import type { TargetScope } from "./scope.types";
+import { ScopeHandler } from "./scopeHandler.types";
+
+export default class OneOfScopeHandler implements ScopeHandler {
+  private scopeHandlers: ScopeHandler[] = this.scopeType.scopeTypes.map(
+    (scopeType) => {
+      const handler = getScopeHandler(scopeType, this.languageId);
+      if (handler == null) {
+        throw new Error(`No available scope handler for '${scopeType.type}'`);
+      }
+      return handler;
+    }
+  );
+
+  public iterationScopeType: OneOfScopeType = {
+    type: "oneOf",
+    scopeTypes: this.scopeHandlers.map(
+      ({ iterationScopeType }) => iterationScopeType
+    ),
+  };
+
+  constructor(
+    public readonly scopeType: OneOfScopeType,
+    private languageId: string
+  ) {}
+
+  /** Return smallest target scope touching position */
+  getScopesTouchingPosition(
+    editor: TextEditor,
+    position: Position,
+    ancestorIndex?: number
+  ): TargetScope[] {
+    if (ancestorIndex !== 0) {
+      // FIXME: Maybe we could find a way to support this opne in the future.
+      throw new NotHierarchicalScopeError(this.scopeType);
+    }
+
+    return keepOnlyBottomLevelScopes(
+      this.scopeHandlers.flatMap((scopeHandler) =>
+        scopeHandler.getScopesTouchingPosition(editor, position, ancestorIndex)
+      )
+    );
+  }
+
+  /** Return all scopes overlapping range not contained by another scope */
+  getScopesOverlappingRange(editor: TextEditor, range: Range): TargetScope[] {
+    return keepOnlyTopLevelScopes(
```

Comment:
I can't decide if this is right. I think we may want to make it so that intersecting scopes only count if they start or end within the range. Then I think this one would be right. Let's discuss tomorrow or Tues